command/agent/testagent.go

```diff
@@ -255,12 +259,13 @@ func (a *TestAgent) start() (*Agent, error) {
 	}
 
 	// Setup the HTTP server
-	http, err := NewHTTPServer(agent, a.Config)
+	httpServers, err := NewHTTPServers(agent, a.Config)
 	if err != nil {
 		return agent, err
 	}
-
-	a.Server = http
+	// TODO: change type or handle better
+	a.Servers = httpServers
+	a.Server = &httpServers[0]
```

Comment:
Are there tests that are specifically looking for `a.Server` that can't be refactored?

Reply:
Looked into doing this, there are a ton of different files that are looking for this. I made a dent in updating them, but about an hour in

	modified:   command/agent/acl_endpoint_test.go
	modified:   command/agent/agent_endpoint_test.go
	modified:   command/agent/alloc_endpoint_test.go
	modified:   command/agent/csi_endpoint_test.go
	modified:   command/agent/deployment_endpoint_test.go
	modified:   command/agent/eval_endpoint_test.go
	modified:   command/agent/event_endpoint_test.go
	modified:   command/agent/fs_endpoint_test.go
	modified:   command/agent/helpers_test.go
	
and there are still 5 or so more files that require turning s.Server into a for loop. Happy to keep trying, but it'll greatly increase the size of the PR

Reply:
Eh, you're probably right that it's better to leave it then. So long as we have at least one set of tests that exercises multiple servers I'm ok with that.

Reply:
Created one test https://github.com/hashicorp/nomad/pull/11582/files#diff-e950133dfdce118008e42fa60a1930583b929388309ad9e4ba2283b65572ba10R72-R89. Let me know if thats what you're looking for.

I left the existing tests as a fork to my branch if I find some willpower I'll see if I can add it as a seperate PR, but can't promise anything.